services/volumesrv/volume.go

```diff
@@ -8,16 +8,17 @@ import (
 
 type Store interface {
 	VolumeAdd(projectID string, volume types.Volume) error
+	VolumeList(projectID string) ([]types.Volume, error)
 	VolumeGet(projectID, idOrName string) (types.Volume, error)
-	VolumeDelete(id string)
+	VolumeDelete(id string) error
 	VolumeUpdate(id string, volume types.Volume) error
 }
 
 type Driver interface {
-	VolumeCreate(ctx context.Context, name string, size int)
-	VolumeDelete(ctx context.Context, id string)
+	VolumeCreate(ctx context.Context, vol types.Volume) error
+	VolumeDelete(ctx context.Context, id string) error
 	VolumeGet(ctx context.Context, id string) (types.Volume, error)
 	VolumeProvider() types.Provider
 	VolumeDriver(ctx context.Context) string
-	VolumeResize(ctx context.Context, id string, size int) error
+	VolumeUpdate(ctx context.Context, vol types.Volume) error
```

Comment:
i've updated this to be more consistent with the store and simplified volume create to take the volume that we expect to be created

Reply:
I don't know if this is a good idea. What if we add the `CreatedAt` and `UpdatedAt` types to the volume? How will those be defined before the volume is created? The current way of doing things (with IDs) allows you to delegate that responsibility to the store. Anytime you need the source of truth for a volume, you ask the store. This breaks that paradigm.

Reply:
okay sure but then how do we volume get on the driver and return a volume type?

Reply:
I guess we don't. We do more specific operations, like `VolumeGetSize(id string)`. The store is the source of truth for for any data that's persistent, while the driver gets more realime data. For example, if you want to update how much of the volume is used, the driver would have a `VolumeGetUsed(id string)` method. 

The driver should ideally not be tied to business logic but I agree there might not be a clean separation.

Reply:
yeah, the lack of this separation of concerns is what part motivated this approach but also made it tricky, what i think we need here is a Go equivalent of a Typescript Partial type.